app/controllers/publishers/vacancies/job_applications_controller.rb

```diff
@@ -78,6 +79,19 @@ def pre_interview_checks
     @reference_requests = @job_application.referees.filter_map(&:reference_request)
   end
 
+  def update_all
+    form_params = params.require(:publishers_job_application_offer_decline_date_form)
+                          .permit(:origin, :offered_at, :declined_at, :status, job_applications: [])
+    form_params[:job_applications] = vacancy.job_applications.where(id: Array(form_params[:job_applications]).compact_blank)
+    @form = Publishers::JobApplication::OfferDeclineDateForm.new(form_params)
+    if @form.valid?
+      vacancy.job_applications.where(id: @form.job_applications).find_each { it.update!(@form.attributes) }
+      redirect_to organisation_job_job_applications_path(vacancy.id, anchor: @form.origin)
+    else
+      render(@form.status == "offered" ? "offered_date" : "declined_date")
+    end
+  end
```

Comment:
almost wondering if this code should be somewhere else? 'update_all' doesn't feel like a good name - were updating offered/declined dates. Maybe it needs its own controlller?